### ISSUE TITLE
Issuing wildcard certificates stopped working

### ISSUE THREAD

**Issue by a123b** [acmesh-official/acme.sh#3036]
It seems that somewhere within the last 3 months Let's Encrypt started requiring a separate TXT record for the wildcard alt domain even if it's the same domain as the main domain.

This causes acme.sh's issuing procedure to fail, here's my `debug 2` log file as an example: https://pastebin.com/dJQMvvv8
This is the same problem @kinslayer1982 described in https://github.com/acmesh-official/acme.sh/issues/2230#issuecomment-654284748.
#3016 might also be a duplicate, but it's not including any relevant logs so I'm not sure...

So what happens is:
- TXT record is set for main domain
- TXT record is set for wildcard "alt" domain (which is the same as the main domain in this case), **overwriting** the previously set TXT record
- acme.sh waits for the first TXT record to propagate, which obviously never happens as it has just been overwritten by the second TXT record
- Times out after 20 mins and fails

Idea how to fix:
- Make new variable `$remaining_dns_entries` which is initially a copy of `$dns_entries`
- For each unique `txtdomain` in `$remaining_dns_entries`, remove the entry and add the TXT record
- Already request validation from Let's Encrypt even if there are still entries in `$remaining_dns_entries`
- Repeat until `$remaining_dns_entries` is empty

**Comment by a123b** [comment]
### Quick fix for those who just need to renew their cert quickly:
Insert a `break` after [this line](https://github.com/acmesh-official/acme.sh/blob/4daef5299172968f258d55932020c0b79389a9e6/acme.sh#L4286) and run the renewal twice.

**Comment by Neilpang** [comment]
did you read this?

>> In the defenition for the host make sure to set the checkbox for "Wildcard" and for "TXT".


https://github.com/acmesh-official/acme.sh/wiki/dnsapi#74-use-ddnssde-api

**Comment by a123b** [comment]
Of course. As I said, it used to work until somewhere around last month. Also, if the dns provider configuration was the problem, I don't think my "quick fix" would have worked...

**Comment by Neilpang** [comment]
@mod242   can you please take a look ?

**Comment by mod242** [comment]
Hello, unfortunately I was away (without Internet) that's why I just had a look right now. Somehow my test-environment seems to behave differently here for ddnss.de.

I get an "...is already verified, skip dns-01." for the second domain. Can you please confirm that you are still seeing this issue with ddnss.de wildcard domains?

**Comment by alexiri** [comment]
I have this issue with duckdns.org domains as well.

**Comment by xobotun** [comment]
Same problem here. Wildcard domain TXT entry overwrites normal domain TXT record:
```
...
Creating domain key
The domain key is here: /acme.sh/xobotun.com/xobotun.com.key
Multi domain='DNS:xobotun.com,DNS:*.xobotun.com'
Getting domain auth token for each domain
Getting webroot for domain='xobotun.com'
Getting webroot for domain='*.xobotun.com'
Adding txt value: aTN5ATzaMJccXnBvGRVCSDfxEeQEuzjmLHRF1rGD2VE for domain:  _acme-challenge.xobotun.com
The txt record is added: Success.
Adding txt value: OiHjgBTPM61zrfSDtQpbc81HdL_sgoyQm2k3ESrAOqo for domain:  _acme-challenge.xobotun.com
The txt record is added: Success.
Let's check each DNS record now. Sleep 20 seconds first.
Checking xobotun.com for _acme-challenge.xobotun.com
Not valid yet, let's wait 10 seconds and check next one.
Checking xobotun.com for _acme-challenge.xobotun.com
...
```
When I looked up DNS entries manually, there was only the second one, obviously.

Will temporarily try to fix by issuing two separate certificates, one for main domain and the other one for wildcard.

**Comment by mod242** [comment]
As I mentioned in #2230 I think this is related to some change in LEs behaviour and not to the API-Implementation for the different providers. Your experience with Yandex seems to increase this impression.

**Comment by OnkelM** [comment]
this bug is still there and has never been adressed, is anyone working on this?
the same issue with overwritten TXT entry because of wildcard happens also on Kasserver (dns_kas.ah)

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.

**Comment by neilpang** [comment]
Not a bug in acme.sh: it adds both TXT values correctly; providers like DuckDNS/ddnss/Yandex only store one TXT record per name and overwrite the first, which is a provider API limitation.
